### ISSUE TITLE
Pause when not watching

### ISSUE THREAD

**Issue by gonzo90017** [fnord12/script.pseudotv#10]
[kodi.old.log](https://github.com/fnord12/script.pseudotv/files/4750747/kodi.old.log)
Windows 7 64bit Kodi 18.7 Pseudo TV 4.5.1

TMNT channel
Advance Settings: 
Force Resume Mode
Pause When not watching
Play Shows in Order

Doesn't matter what channel i'm on. If I press X to close PseudoTV. Launch Pseudotv again and go to the TMNT channel it plays exactly where I left off. Let's say S1:E5

Doesn't matter what channel i'm on. If I press ESC then yes to exit PseudoTV. Launch Pseudotv again an go to the TMNT channel. The channel goes back S1:E1

I can reproduce this issue every time. So it seems like when exiting using ESC something is going on that it's not saving or closing properly.  I uploaded my log. If it's too big you can delete it and i'll reproduce both the working exit and the non working exit so the file won't be so big.

**Comment by fnord12** [comment]
The log is great, thanks.

A couple questions:
1. Are you saying this is new behavior with 4.5.1?  Because this sounds like the experience we've had with Paused channels since 17 (e..g  https://github.com/Steveb1968/script.pseudotv/issues/5 ).  

2. Can you find your settings2.xml file in \userdata\addon_data\script.pseudotv.  Look at all the values for this channel when you quit with X.  And see if anything is different when you quit with ESC.  Especially the _time value.  

My experience has been that the time value is correctly written both times, and the reset problem isn't (directly?) related to how you quit.  IOW it's not failing because it's incorrectly writing something on quit, it's failing because of whatever is causing the channel reset problem that's been plaguing us.  Although the consistent difference between ESC and Quit is worth pulling at.  I'll try to investigate that further.  But unless you tell me that you're seeing new behavior and i need to revert something, no promises on the timing.

**Comment by fnord12** [comment]
FYI, it's not the issue but i see you're in Realtime mode overall and you also have an Advanced rule to set the music video channel to Realtime.  That's at best unnecessary as far as i know; the point of those rules at the channel level is to override the overall mode.  Is there a reason you did that?

And it's also set to Reset Every Day, is that because you keep the overall reset setting to Never and this allows you to add more videos without having to do a reset?  

Nothing here relates to your problem; it just helps me to understand how people use PSTV.

**Comment by gonzo90017** [comment]
It's the same issue that's been around for a while. 

Did some more testing and noticed something I didn't notice yesterday. The TMNT channel only resets to S1:E1 if I exit while watching that channel. If i'm on my Rock channel and exit using ESC there. When I return to PseudoTV the TMNT channel is where I left off the last time.

Here are my settings.xml findings:

After Exit Using X:  <setting id="Channel_6_time" value="6100" />

After Exit using ESC:  <setting id="Channel_6_time" value="6104.25300598" />

This is where it gets interesting. This is the settings.xml time value when I reopen Pseudotv after having Exited using ESC:   <setting id="Channel_6_time" value="0" />

[Settings.zip](https://github.com/fnord12/script.pseudotv/files/4753354/Settings.zip)

**Comment by gonzo90017** [comment]
> FYI, it's not the issue but i see you're in Realtime mode overall and you also have an Advanced rule to set the music video channel to Realtime. That's at best unnecessary as far as i know; the point of those rules at the channel level is to override the overall mode. Is there a reason you did that?
> 
> And it's also set to Reset Every Day, is that because you keep the overall reset setting to Never and this allows you to add more videos without having to do a reset?
> 
> Nothing here relates to your problem; it just helps me to understand how people use PSTV.

I forgot I had set the Rock channel to Real time mode. I thought I encountered a bug but it wasn't  anything. I was just putting 4.5.1 through its paces. 

The reason I have the Rock channel reset every day is because i only have around 19 hours worth of content. And since I have Pseudotv to reset channels to Never. With this setting PseudoTV resets the channel right before launching it, instead of in the background after it's launched. And I also add more videos whenever I have a chance :)

**Comment by fnord12** [comment]
Yeah, what you're describing sounds like the standard Reset Problem as i know it.  That's why the first fix was to introduce that startup channel, so your other channels aren't affected.  What you're seeing in the settings2 file makes sense, too.  As far as the code is concerned, it's correctly keeping the seektime (not setting back to 0) and if you look in your debug log after a fail/reset, you'll see that it correctly sets the seektime.  Something mysterious (not shown in the log) happens after that which causes the playlist to reset.

I'm not suggesting you do this (it wouldn't help with anything, it's just proof of the problem), but if you were to reverse your mode setup, so that the overall mode was Resume and the rest were forced to Realtime, and then you made the code change mentioned in the Release Notes, you'd see that the TMNT channel was now not resetting if it's the startup channel, and all the others would.  Although i'm not sure how the Pause option would affect things.

It's weird how i can fix it for one mode or the other, but putting in a simple If statement causes it to fail.

**Comment by gonzo90017** [comment]
[kodi.zip](https://github.com/fnord12/script.pseudotv/files/4754430/kodi.zip)

[Overlay.zip](https://github.com/fnord12/script.pseudotv/files/4755123/Overlay.zip)

Tried your suggestion. I think I did everything right. 
Modified Overlay.py, Changed overall mode to Resume, I removed all of the Advanced channel settings for the TMNT channel just to make sure it didn't affect the results. Set the startup channel to 6 (TMNT).

But still the same behavior as above. Exiting using X works fine. Exiting with ESC and the channel resets. Just thought I should let you know since it's been a while since anybody has looked into this issue. 

Forgot to put kodi in debug mode. It's easy to reproduce so let me know if you would like me to do that.

**Comment by fnord12** [comment]
I've been using the current 5.0.1. release in a Resume Mode/Pause When Not Watching state for a while now and it's working great for me so I'm closing this old issue.
